vispy/app/inputhook.py

```diff
@@ -53,23 +50,24 @@ def enable(self, app=None):
         """
 
         from .. import app as _app
+
         self.app = app or _app.use_app()
         self.manager.set_inputhook(self._vispy_inputhook)
         return app
 
     def _vispy_inputhook(self):
         try:
-            t = time()
             while not stdin_ready():
                 self.app.process_events()
 
-                used_time = time() - t
-                if used_time > 10.0:
-                    sleep(1.0)
-                elif used_time > 0.1:
-                    sleep(0.05)
-                else:
-                    sleep(0.001)
+                # for more context.
+                # we need to wait out on the event loop to prevent CPU stress
+                # but not wait too much, to maintain fluidity.
+                # refer https://github.com/vispy/vispy/issues/945
+                def sleep_duration_sec():
+                    return 0.05
+                self.app.sleep(sleep_duration_sec())
```

Comment:
I wanted to document the constant rather than just pass one, but now that I think about it, using a named argument is nicer anyway. 
(Python has no way to declare a variable as a constant, so I used to do that on my personal projects for things like hardcoded strings)